services/spam-ban/spam-banning.service.js

```diff
@@ -0,0 +1,111 @@
+const { EmbedBuilder } = require('discord.js');
+const { isAdmin } = require('../../utils/is-admin');
+const config = require('../../config');
+
+class SpamBanningService {
+  static async handleInteraction(interaction) {
+    const message = interaction.options.getMessage('message');
```

Comment:
I'm assuming we'll be calling this command only inside the automod channel. We'll add this condition in our discord server's setting. Though a sanity check here won't hurt. Can we add a guard here to check if this command is only called in that channel?